model-optimizer/mo/utils/ir_reader/layer_to_class.py

```diff
@@ -342,6 +342,39 @@ def copy_graph_with_ops(graph: Graph) -> Graph:
 
     # Nodes postprocessing stage in new graph
     for op in new_graph.get_op_nodes():
+        # tensor names restoring
+        debug_dict = {}
+
+        for out_port in op.ports:
+            out_tensor_names = op.ports[out_port][1]
+
+            # handle Constant operations with old style output port numbering
+            if op.soft_get('type') == 'Const':
+                assert len(op.ports) == 1, 'Something wrong with Constant node: {}, wrong number ' \
+                                           'of output ports: {}!'.format(op.soft_get('name'), len(op.ports))
+                if out_port == 1:
+                    out_port = 0
+
+            debug_dict[out_port] = []
+            if out_tensor_names is not None:
+                # handle tensor names with commas and add them to dictionary as separated
+                if out_tensor_names.find(',') >= 0:
+                    out_tensor_names = (out_tensor_names.replace('\\,', '<comma_in_tensor_name>')).split(',')
+                    for out_tensor_name in out_tensor_names:
+                        debug_dict[out_port].append((out_tensor_name, out_port,
+                                                     out_tensor_name.replace('<comma_in_tensor_name>', ',')))
+                else:
+                    debug_dict[out_port] = [(out_tensor_names, out_port, out_tensor_names)]
+            else:
+                debug_dict[out_port] = None
+
+        # provide tensor names to data nodes
+        if len(debug_dict) > 0:
+            for out in op.out_nodes():
+                if debug_dict[out + len(op.in_nodes())] is not None:
+                    op.out_node(out)['fw_tensor_debug_info'] = debug_dict[out + len(op.in_nodes())]
+
+        # operations postprocessing with some special types
```

Comment:
Let's move new functionality to a separate function. Current function is too long already.

Reply:
Ok, done.